web/modules/custom/joinup_licence/js/search_auto_submit.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * @file
+ * Autocomplete additional behaviors for invite forms.
+ */
+
+(function ($, Drupal) {
+
+  /**
+   * Attaches the auto submit enhancements to the licence search select list.
+   *
+   * @type {Drupal~behavior}
+   *
+   * @prop {Drupal~behaviorAttach} attach
+   *   Attaches the auto submit behaviors.
+   */
+  Drupal.behaviors.searchAutoSubmit = {
+    attach: function (context) {
+      $('.auto_submit', context).on('change', function (event) {
+        var $value = $(event.target).val();
+
+        // In order to ensure that the original values are maintained or at
+        // least, that no malformed string is passed, check if the value
+        // selected matches the same pattern as the requirements of the route.
+        // Note: The + symbol is maintained here compared to the route
+        // requirements where it is assumed that it has been converted to a
+        // space character.
+        var pattern = /^[a-zA-Z0-9][a-zA-Z0-9.\+-]+$/;
+        if (pattern.test($value)) {
```

Comment:
It is easy to happen. This is not an AJAX request that everything goes through the server. Simply editing the option's value in the source code before selecting the value will pass that string to the function.
Of course, I cannot tell you if a specific string passed could cause harm. I would be a hacker if I new..

Reply:
OK, so what happens if such a string is passed to the URL? What the param converter will do if the pattern (which, by the way is taken from the route check) is not fulfilled?

Reply:
No, I don't think that we are in danger of having XSS vulnerability in the route parameter. I was mostly considering cases where a JS was somehow put into there altering the general JS command. I am not really expert in JS so I don't know if this can be used for other purposes. I will remove it.